socorro/external/es/crashstorage.py

```diff
@@ -158,8 +160,43 @@ def truncate_keyword_field_values(fields, data):
         if isinstance(value, str) and len(value) > MAX_KEYWORD_FIELD_VALUE_SIZE:
             data[field_name] = value[:MAX_KEYWORD_FIELD_VALUE_SIZE]
 
-            # FIXME(willkg): When we get metrics throughout the processor, we should
-            # keep track of this with an .incr().
+
+def truncate_string_field_values(fields, data):
+    """Truncates string field values greater than MAX_STRING_FIELD_VALUE_SIZE length
+
+    Note: This modifies the data dict in-place and only looks at the top level.
+
+    :arg dict fields: the super search fields schema
+    :arg dict data: the data to look through
+
+    """
+    string_fields = [
+        field
+        for field in fields.values()
+        if (field.get("storage_mapping") or {}).get("type", "") == "string"
+    ]
```

Comment:
This will pick up keyword values, too. I didn't see a good way of building a set of non-overlapping fields we want to operate on with what we have now. When we redo how Elasticsearch crash storage works, we can clean this whole thing up. Until then, this is a hack-ish thing.